sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -61,6 +61,7 @@ class JacksonParser(
     dt match {
       case st: StructType => makeStructRootConverter(st)
       case mt: MapType => makeMapRootConverter(mt)
+      case at: ArrayType => makeArrayRootConverter(at)
```

Comment:
> This change accepts the json datasource form that the master can't parse?

Right, it accept arrays of any types comparing to the master which accepts arrays of structs only

> If so, I think we need tests in JsonSuite ...

I added a few tests to `JsonSuite` or do you mean some concrete test case?

Reply:
You've already added tests in `JsonSuite`? It seems there are tests in `JsonFunctionsSuite`, `JsonExpressionSuite`, and `SQLQueryTestSuite` now?

Reply:
Sorry, I didn't catch you meant specific class. Just in case, what is the reason for adding tests to `JsonSuite`? I changed behavior of a function, so, `JsonFunctionsSuite` is perfect place for new tests, I believe. At the moment, you cannot specify a schema of json objects different from `StructType` in `DataFrameReader`. In this way, the `ArrayType` can come into `JacksonParser` only from json functions (`from_json`). I can add similar test to `JsonSuite` as in `JsonFunctionsSuite` but it doesn't make any sense from my point of view.

Reply:
ah, ok. You touched the the `JacksonParser.scala` file, so I though there were some behaivour changes in the json datasorce. But I notice that there are not, so the current tests are enough. Thanks.